### ISSUE TITLE
Back and fw arrows are disabled (tab history is lost) if FF is force closed while home page is in view

### ISSUE THREAD

**Issue by alinamoldovan84** [mozilla-mobile/firefox-ios#21602]
## Steps to reproduce
1. Open a new tab and open the following web pages on the same tab: Wikipadia, Homepage, x.com 
1. Long tap the back arrow button and choose homepage (link under the x.com)
1. Both back and fw arrow buttons are enabled 
1. Force close the device 
1. Open the FF app again and check the arrow buttons 

### Expected behavior
Tab history is persisted. The arrow buttons are enabled

### Actual behavior
Tab history is lost. The arrow buttons are disabled

### Device & build information
* Device: iPhone 15+ (17.5.1), iPhone 15 (17.5.1) 
* Build version: v130 (44542) 
* First seen version:? 

### Attachments: 

https://github.com/user-attachments/assets/d6a5a1f9-1727-460a-866b-a134e26e8e38

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-9839)

**Comment by data-sync-user** [comment]
➤ Matt Reagan commented:

Note: this appears to be a very longstanding bug. In trying to determine when this began happening I’ve been rolling back to previous commits and so far I’ve rolled all the way back to March 19th and I am still able to replicate the issue. (cc Norberto Andres Furlan)

**Comment by data-sync-user** [comment]
➤ Matt Reagan commented:

This looks like it is a result of us simply not loading the session data specifically when the current tab is on the homepage. The logic appears to have been added deliberately and has been in place for quite a while. Doing a bit more digging to see if I can understand why it was originally added; removing the check for the homepage appears to fix this bug. (cc Norberto Andres Furlan )

**Comment by data-sync-user** [comment]
➤ Matt Reagan commented:

QA note: merged to main (v131) and also backported for 130.1 release.

**Comment by data-sync-user** [comment]
➤ Andrei Bodea commented:

Hi, Matt Reagannote I re-checked this on v9000 (44732) with iPhone 15 Pro (17.5.1), iPhone 15 (17.5), and iPhone SE (18.0).

We did over 10 tries and the history persisted and we were able to navigate back and forth (even after a restart). 

Started: Homepage > Youtube > Homepage > Facebook, when I selected (was the middle homescreen I opened after YouTube) So that’s why there are 4 displayed.

[^RPReplay_Final1725019295.MP4]

This issue will remain with QA Needed (but it's verified in v9000).
{{The status will be changed when an official build for 131 or 130.1 }}is{{ sent to QA.}}
